dissect/target/filesystem.py

```diff
@@ -948,10 +951,10 @@ def __init__(self, **kwargs):
         self.root = VirtualDirectory(self, "/")
 
     @staticmethod
-    def detect(fh):
+    def detect(fh: BinaryIO):
```

Comment:
Add type hints to `Virtual*` and return types?

Reply:
I decided against type hinting the other `Virtual*` classes as they were barely touched and would be too much scope creep.